ingest/processors/csvrouter.go

```diff
@@ -0,0 +1,153 @@
+/*************************************************************************
+ * Copyright 2022 Gravwell, Inc. All rights reserved.
+ * Contact: <legal@gravwell.io>
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD 2-clause license. See the LICENSE file for details.
+ **************************************************************************/
+
+package processors
+
+import (
+	"bytes"
+	"encoding/csv"
+	"errors"
+	"fmt"
+
+	"github.com/gravwell/gravwell/v3/ingest/config"
+	"github.com/gravwell/gravwell/v3/ingest/entry"
+)
+
+const (
+	CSVRouterProcessor = `csvrouter`
+)
+
+var (
+	ErrInvalidColumnIndex = errors.New("Invalid column index")
+)
+
+type CSVRouteConfig struct {
+	Route_Extraction int
+	Route            []string
+	Drop_Misses      bool
+}
+
+type CSVRouter struct {
+	nocloser
+	CSVRouteConfig
+	routes map[string]entry.EntryTag
+	drops  map[string]struct{}
+}
+
+func CSVRouteLoadConfig(vc *config.VariableConfig) (c CSVRouteConfig, err error) {
+	err = vc.MapTo(&c)
+	return
+}
+
+func NewCSVRouter(cfg CSVRouteConfig, tagger Tagger) (*CSVRouter, error) {
+	cr := &CSVRouter{}
+	if err := cr.init(cfg, tagger); err != nil {
+		return nil, err
+	}
+	return cr, nil
+}
+
+func (cr *CSVRouter) Config(v interface{}, tagger Tagger) (err error) {
+	if v == nil {
+		err = ErrNilConfig
+	} else if cfg, ok := v.(CSVRouteConfig); ok {
+		err = cr.init(cfg, tagger)
+	} else {
+		err = fmt.Errorf("Invalid configuration, unknown type type %T", v)
+	}
+	return
+}
+
+func (cr *CSVRouter) init(cfg CSVRouteConfig, tagger Tagger) (err error) {
+	var rts []route
+	if rts, err = cfg.validate(); err != nil {
+		return
+	}
+	cr.CSVRouteConfig = cfg
+	cr.routes = make(map[string]entry.EntryTag)
+	cr.drops = make(map[string]struct{})
+	for _, r := range rts {
+		if r.drop {
+			cr.drops[r.val] = empty
+		} else {
+			var tg entry.EntryTag
+			if tg, err = tagger.NegotiateTag(r.tag); err != nil {
+				err = fmt.Errorf("Failed to get tag %s for %s: %v", r.tag, r.val, err)
+				return
+			}
+			cr.routes[r.val] = tg
+		}
+	}
+	return
+}
+
+func (cr *CSVRouter) Process(ents []*entry.Entry) (rset []*entry.Entry, err error) {
+	if len(ents) == 0 {
+		return
+	}
+	rset = ents[:0]
+	for _, ent := range ents {
+		if ent == nil {
+			continue
+		} else if ent = cr.processItem(ent); ent != nil {
+			rset = append(rset, ent)
+		}
+	}
+	return
+}
+
+func (cr *CSVRouter) processItem(ent *entry.Entry) *entry.Entry {
+	r := csv.NewReader(bytes.NewReader(ent.Data))
+	r.FieldsPerRecord = -1
```

Comment:
i think we need to put in `r.LazyQuotes=true` and `r.TrimLeadingSpace=true` here